packages/@aws-cdk/aws-vpclattice-alpha/lib/service-network.ts

```diff
@@ -0,0 +1,150 @@
+
+  /**
+   * Whether to enable sharing of the service network
+   *
+   * @default false
+   */
+  readonly enableSharing?: boolean;
+
+  /**
+   * Tags for the service network
+   *
+   * @default - No tags
+   */
+  readonly tags?: string;
+
+  /**
+   * Services to associate with the service network
+   *
+   * @default - No services
+   */
+  readonly services?: IService[];
+}
+
+/**
+ * Attributes for importing an existing VPC Lattice service network
+ */
+export interface ServiceNetworkAttributes {
+  /**
+   * The ARN of the service network
+   */
+  readonly serviceNetworkArn: string;
+
+  // TODO: restore ID? Constructing the ARN from the ID will need accountId, region, etc
+  // readonly serviceNetworkId?: string;
+}
+
+/**
+ * A VPC Lattice service network
+ */
+export class ServiceNetwork extends ServiceNetworkBase {
+  /**
+   * Import an existing VPC Lattice service network from its ARN
+   *
+   * @param scope The parent construct
+   * @param id The construct ID
+   * @param serviceNetworkArn The ARN of the service network
+   * @returns A service network construct
+   */
+  public static fromServiceNetworkArn(scope: Construct, id: string, serviceNetworkArn: string): IServiceNetwork {
+    return ServiceNetwork.fromServiceNetworkAttributes(scope, id, { serviceNetworkArn: serviceNetworkArn });
+  }
+
+  /**
+   * Import an existing VPC Lattice service network from its attributes
+   *
+   * @param scope The parent construct
+   * @param id The construct ID
+   * @param attrs The service network attributes
+   * @returns A service network construct
+   */
+  public static fromServiceNetworkAttributes(scope: Construct, id: string, attrs: ServiceNetworkAttributes): IServiceNetwork {
+    class UnownedServiceNetworkReference extends ServiceNetworkBase {
+      public readonly serviceNetworkArn: string = attrs.serviceNetworkArn;
+      // Extract the serviceNetworkId from the ARN, which has the format `arn:aws:vpc-lattice:<region>:<account>:servicenetwork/<service-network-id>`
+      public readonly serviceNetworkId: string = Stack.of(scope).splitArn(attrs.serviceNetworkArn, ArnFormat.SLASH_RESOURCE_NAME).resourceName!;
+    }
+    return new UnownedServiceNetworkReference(scope, id);
+  }
+
+  /**
+   * The ARN of the service network
+   * @attribute
+   * @returns a token representing the service network ARN
+   */
+  public readonly serviceNetworkArn: string;
+
+  /**
+   * The ID of the service network
+   * @attribute
+   * @returns a token representing the service network ID
+   */
+  public readonly serviceNetworkId: string;
+
+  /**
+   * Creates a new VPC Lattice service network
+   *
+   * @param scope The parent construct
+   * @param id The construct ID
+   * @param props The service network properties
+   */
+  constructor(scope: Construct, id: string, props: ServiceNetworkProps) {
+    super(scope, id);
+
+    // TODO: validation?
+
+    // TODO: tags
+
+    const resource = new CfnServiceNetwork(this, id, {
```

Comment:
```suggestion
    const resource = new CfnServiceNetwork(this, `Resource`, {
```

Always use `Resource` for the default/main resource.
https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md#construct-ids